taichi/backends/vulkan/aot_module_builder_impl.cpp

```diff
@@ -46,9 +105,10 @@ uint32_t AotModuleBuilderImpl::to_vk_dtype_enum(DataType dt) {
 
 void AotModuleBuilderImpl::write_spv_file(
     const std::string &output_dir,
-    const TaskAttributes &k,
+    TaskAttributes &k,
     const std::vector<uint32_t> &source_code) const {
   const std::string spv_path = fmt::format("{}/{}.spv", output_dir, k.name);
+  k.source_path = spv_path;
```

Comment:
Can we directly store `spv_path` to `aot::CompiledOffloadedTask`, so that `k` remains const?

Reply:
@k-ye You mean doing it inside `aot::CompiledOffloadedTask AotDataConverter::visit(const TaskAttributes &in)` function right? I thought about it but didn't went down that path mainly because: 
we "need/want" the `output_dir` information in the `spv_path` but I cannot convince myself passing it to converter... (converter should be a pure data structure transformation to me and should have nothing to do with where you dump the files...

Reply:
One dump solution: separate path generation from `write_spv_file`..

Reply:
sorry I was dumb :P should be working now